cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseDDLExecutorFactory.java

```diff
@@ -46,7 +48,13 @@ public HBaseDDLExecutor get() {
       // HBase DDL executor extension is provided.
       executor = extensions.values().iterator().next();
     } else {
-      // HBase DDL executor extension is not provided. Return the version specific executor instance.
+      if (!extensionDir.isEmpty()) {
```

Comment:
If that's the requirement, I think it's better to implement this as a master startup check instead of in here.